modules/src/clients/ics07_tendermint/client_state.rs

```diff
@@ -167,6 +167,41 @@ impl ClientState {
             clock_drift: self.max_clock_drift,
         })
     }
+
+    /// Verify the time and height delays
+    pub fn verify_delay_passed(
+        current_time: Timestamp,
+        current_height: Height,
+        processed_time: Timestamp,
+        processed_height: Height,
+        delay_period_time: Duration,
+        delay_period_blocks: u64,
+    ) -> Result<(), Error> {
+        let time = (processed_time + delay_period_time).map_err(Error::timestamp_overflow)?;
+        if !(current_time == time || current_time.after(&time)) {
+            return Err(Error::not_enough_time_elapsed());
```

Comment:
It could be useful to augment the error type here so that we also capture the specific time parameters that caused the error. Otherwise it will be difficult to debug later on. The return statement would look as follows:

```rust
return Err(Error::not_enough_time_elapsed(current_time, earliest_time));
```

Reply:
5c1ee1d829aa598f0c649ef4534cf6de08d083ab